### PR TITLE
chore(sandbox): Use wasmer in embedded executor

### PR DESCRIPTION
**Pull request by ark0f**
About to replace #4267

### CODE REVIEW

**Review by breathx — CHANGES_REQUESTED** [review]
To be aligned as discussed

File: lazy-pages/common/src/lib.rs [hunk 1]
```diff
@@ -74,7 +74,7 @@ pub struct GlobalsAccessError;
 /// Globals access trait.
 pub trait GlobalsAccessor {
     /// Returns global `name` value, if `name` is I64 global export.
-    fn get_i64(&self, name: &LimitedStr) -> Result<i64, GlobalsAccessError>;
+    fn get_i64(&mut self, name: &LimitedStr) -> Result<i64, GlobalsAccessError>;
```
Comment (ByteNacked) [inline lazy-pages/common/src/lib.rs:77]: Is it intended change?

File: sandbox/sandbox/build.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{env, fs, path::PathBuf};
+
+fn main() {
+    let out_dir = env::var("OUT_DIR").unwrap();
+    let out_dir = PathBuf::from(out_dir);
+    fs::write(out_dir.join("placeholder"), "placeholder file").unwrap();
+}
```
Comment (ByteNacked) [inline sandbox/sandbox/build.rs:25]: Could you please add a comment explaining where and what for this placeholder file is used?

File: sandbox/sandbox/src/embedded_executor.rs [hunk 3]
```diff
@@ -24,98 +24,247 @@ use crate::{
...
 
     fn data_mut(&mut self) -> &mut Self::State {
-        self.0.data_mut()
+        &mut self.state.as_mut(&mut self.inner).inner
     }
 }
 
 impl<T> AsContext for Store<T> {}
 
 /// wasmi caller wrapper.
```
Comment (ByteNacked) [inline sandbox/sandbox/src/embedded_executor.rs:244]: wasmi -> wasmer

File: sandbox/sandbox/src/embedded_executor.rs [hunk 4]
```diff
@@ -465,8 +624,8 @@ fn to_wasmi(value: Value) -> RuntimeValue {
     match value {
```
Comment (ByteNacked) [inline sandbox/sandbox/src/embedded_executor.rs:624]: to_wasmi -> to_wasmer

**Review by ByteNacked — APPROVED** [review]
GJ! LGTM

File: lazy-pages/common/src/lib.rs (hunk 1 above)
Comment (ark0f) [reply lazy-pages/common/src/lib.rs:77]: Unfortunately yes. Globals acquiring requires mutable store with `wasmer`

File: ethexe/processor/src/host/mod.rs [hunk 5]
```diff
@@ -54,8 +54,6 @@ pub(crate) struct InstanceCreator {
 
 impl InstanceCreator {
     pub fn new(runtime: Vec<u8>) -> Result<Self> {
-        gear_runtime_interface::sandbox_init();
```
Comment (breathx) [inline ethexe/processor/src/host/mod.rs:57]: why dont we need to init sandbox anymore?

**Review by breathx — APPROVED** [review]

File: ethexe/processor/src/host/mod.rs (hunk 5 above)
Comment (ark0f) [reply ethexe/processor/src/host/mod.rs:57]: The only reason of the function existence is necessity to initialize `wasmer` trap handler before `lazy-pages` one. Now it happens right in `lazy-pages` crate

**Pull request merged**